src/ApiHandler.ts

```diff
@@ -361,47 +361,63 @@ export default class ApiHandler {
 		}
 	}
 
+	/**
+	 *
+	 * @param hash BlockHash to make call at.
+	 * @param stash: _Stash_ address.
+	 */
 	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-	async fetchStakingLedger(hash: BlockHash, address: string) {
+	async fetchStakingInfo(hash: BlockHash, stash: string) {
 		const api = await this.ensureMeta(hash);
 
-		const [header, staking] = await Promise.all([
+		const [header, optionController] = await Promise.all([
 			api.rpc.chain.getHeader(hash),
-			api.query.staking.ledger.at(hash, address),
+			api.query.staking.bonded.at(hash, stash), // Option<AccountId> representing the controller
 		]);
 
 		const at = {
 			hash,
 			height: header.number.toNumber().toString(10),
 		};
 
-		if (staking.isNone) {
-			// Address is not a Controller, no need to look up slashing spans.
+		if (optionController.isNone) {
+			// if bonded.isNone, `stash` address param is not a stash.
```

Comment:
We may actually want to throw a 4xx in this case.

Reply:
For both error comments, I agree. I setup up some rudimentary error handling which bubbles up the message and status code to get.

Do you think we should put in some slightly more robust error handling features in this PR?